src/analyses/libraryFunctions.ml

```diff
@@ -346,6 +346,10 @@ let classify fn exps: categories =
   | "pthread_mutex_unlock" | "__pthread_mutex_unlock" | "up_read" | "up_write"
   | "up" | "pthread_spin_unlock"
     -> `Unlock
+  | "pthread_setspecific" as x ->
+    (if not (GobConfig.get_bool "sem.unknown_function.invalidate.globals") then
+       failwith "pthread_setspecific is only analyzed soundly when sem.unknown_function.invalidate.globals is set.");
+    `Unknown x
```

Comment:
I think it could just add it in the new style as
```ocaml
unknown ~attrs:[InvalidateGlobals] ...
```

Reply:
Is there a way to specify that the arguments needs to be considered escaped though?

Reply:
We haven't generalized thread escapes like that. The escape analysis explicitly does it for thread creation, so the easiest solution is to just add a case for this special function as well.